source/mir/random/flex/internal/area.d

```diff
@@ -0,0 +1,766 @@
+                                          f0(r), f1(r), f2(r));
+            determineSqueezeAndHat(iv);
+            return iv;
+        };
+
+        // test left side
+        auto hs1 = dhs(-1, 1);
+        assert(hs1.hat.slope == 0);
+        assert(hs1.hat.intercept == 1);
+        assert(hs1.squeeze.slope == 2);
+        assert(hs1.squeeze.intercept == -1);
+
+        // test right side
+        auto hs2 = dhs(1, 3);
+        assert(hs2.hat.slope == 4);
+        assert(hs2.hat.intercept == -3);
+        assert(hs2.squeeze.slope == 2);
+        assert(hs2.squeeze.intercept == -1);
+    }
+}
+
+/**
+Flex-specific constants cutoffs for numeric errors.
+*/
+template constants(S)
+{
+    import mir.internal.math: sqrt;
+    enum S smallLog = sqrt(sqrt(S.epsilon * 5));
+    enum S smallExp = sqrt(sqrt(S.epsilon * 120));
+}
+
+alias hatArea(S) = area!(true, S);
+alias squeezeArea(S) = area!(false, S);
+
+/**
+Computes the area below either the hat or squeeze function
+in-between a interval `iv`.
+Based on table 1, 2 and general equation (3) from Botts et al. (2013).
+
+    (F_T(sh(r))- F_T(sh(l))) / sh.slope
+
+Params:
+    isHat = whether to calculate the hat or squeeze area
+    iv = Interval to use
+
+Returns: Computed area below either hat or squeeze.
+*/
+void area(bool isHat, S)(ref Interval!S iv)
+in
+{
+    assert(iv.lx < iv.rx, "invalid interval");
+}
+body
+{
+    import std.math: signbit, frexp, LOG2E, isFinite;
+    import mir.internal.math: copysign, exp, log2, fabs;
+    import mir.random.flex.internal.transformations : antiderivative;
+    import mir.random.flex : flexInverse;
+    enum one_div_3 = S(1) / 3;
+    enum one_div_6 = S(1) / 6;
+    enum one_div_24 = S(1) / 24;
+
+    S area = void;
+
+    static if (isHat)
+        auto sh = iv.hat;
+    else
+        auto sh = iv.squeeze;
+
+    // check difference to left and right starting point
+    // sigma in the paper (1: left side, -1, right side
+    immutable S leftOrRight = (iv.rx - sh.y) > (sh.y - iv.lx) ? 1 : -1;
+    immutable shL = sh(iv.lx);
+    immutable shR = sh(iv.rx);
+    immutable ivLength = iv.rx - iv.lx;
+    auto z = leftOrRight * sh.slope * ivLength;
+
+    // sh.y is the boundary point where f obtains its maximum
+
+    // specializations for T_c family (page 6, table 2 from Botts et al. (2013))
+    if (iv.c == 0)
+    {
+        if (fabs(z) < constants!S.smallExp)
+        {
+            area = exp(sh.a) * (ivLength) * (1 + z * S(0.5) + (z * z) * one_div_6
+                                               + (z * z * z) * one_div_24);
+        }
+        else
+        {
+            area = (exp(shR) - exp(shL)) / sh.slope;
+        }
+    }
+    else
+    {
+        z /= sh.a;
+        auto sgnc = copysign(S(1), iv.c);
+        if (!(sgnc * shL >= 0) || !(sgnc * shR >= 0))
+        {
+            area = S.max;
+        }
+        else if (iv.c == 1)
+        {
+            area = S(0.5) * sh.a * ivLength * (z + 2);
```

Comment:
Coming back to these `sh.a` contributions now that we are clear on the meaning of stuff in your writeup: what was bothering me about these components is that Table 2 explicitly gives the contribution as `f(x_0)` and not `\tilde{f}(x_0)`.  So the immediate relationship to `\alpha` wasn't obvious.